addOns/ratelimit/src/main/java/org/zaproxy/addon/ratelimit/ExtensionRateLimit.java

```diff
@@ -0,0 +1,290 @@
+        HttpSender.addListener(this);
+
+        // As long as we're not running as a daemon
+        if (hasView()) {
+            extensionHook.getHookView().addOptionPanel(getOptionsRateLimitPanel());
+            extensionHook.getHookMenu().addToolsMenuItem(getRateLimitMenuItem());
+            extensionHook.getHookMenu().addPopupMenuItem(getPopupMenu());
+            extensionHook.getHookView().addStatusPanel(getStatusPanel());
+        }
+    }
+
+    @Override
+    public boolean canUnload() {
+        return true;
+    }
+
+    @Override
+    public void unload() {
+        super.unload();
+        HttpSender.removeListener(this);
+    }
+
+    private OptionsRateLimitPanel getOptionsRateLimitPanel() {
+        if (optionsRateLimitPanel == null) {
+            optionsRateLimitPanel = new OptionsRateLimitPanel();
+        }
+        return optionsRateLimitPanel;
+    }
+
+    private ZapMenuItem getRateLimitMenuItem() {
+        if (rateLimitMenuItem == null) {
+            rateLimitMenuItem = new ZapMenuItem("ratelimit.topmenu.tools.title");
+            rateLimitMenuItem.setIcon(ICON);
+            rateLimitMenuItem.addActionListener(
+                    e ->
+                            Control.getSingleton()
+                                    .getMenuToolsControl()
+                                    .options(OptionsRateLimitPanel.PANEL_NAME));
+        }
+        return rateLimitMenuItem;
+    }
+
+    private AbstractPanel getStatusPanel() {
+        if (statusPanel == null) {
+            trackerTableModel = new RateLimitTrackerTableModel();
+
+            ZapTable trackerTable = new ZapTable(trackerTableModel);
+            trackerTable.setColumnSelectionAllowed(false);
+            trackerTable.setCellSelectionEnabled(false);
+            trackerTable.setRowSelectionAllowed(false);
+            trackerTable.setAutoCreateRowSorter(true);
+            trackerTable.setAutoCreateColumnsFromModel(true);
+            trackerTable.setColumnControlVisible(true);
+            trackerTable.setDoubleBuffered(true);
+            trackerTable.setDefaultRenderer(
+                    Date.class, new DefaultTableRenderer(new DateFormatStringValue()));
+
+            trackerTable.getColumnModel().getColumn(0).setMinWidth(200);
+            trackerTable.getColumnModel().getColumn(0).setPreferredWidth(800);
+
+            trackerTable.getColumnModel().getColumn(1).setMinWidth(200);
+            trackerTable.getColumnModel().getColumn(1).setPreferredWidth(800);
+
+            trackerTable.getColumnModel().getColumn(2).setMinWidth(50);
+            trackerTable.getColumnModel().getColumn(2).setPreferredWidth(200);
+
+            trackerTable.getColumnModel().getColumn(3).setMinWidth(50);
+            trackerTable.getColumnModel().getColumn(3).setPreferredWidth(200);
+
+            trackerTable.getColumnModel().getColumn(4).setMinWidth(80);
+            trackerTable.getColumnModel().getColumn(4).setPreferredWidth(250);
```

Comment:
Is it really necessarily to define this somewhat statically like this, or can things just be `pack()`ed before display and workout alright?

Reply:
fixed